generator/generator.py

```diff
@@ -28,6 +28,14 @@ def wire(self, port0, port1):
         assert isinstance(port1, PortReferenceBase)
         self.wires.append((port0, port1))
```

Comment:
Let's do the following, lets sort (port0, port1) based on something (the addresses is fine). So this would look like `self.wires.append((max(port0, port1), min(port0, port1))` so there is always an order. Then in `remove_wire` we only need to look for `(max(port0, port1), min(port0, port1))`.

Also, in `wire()` we should check that this connection has not already been made. If it has then we should just print a warning and return (without an error). This is to avoid doing 

```python
wire(a, b)
wire(a, b)
```

which is valid right now. If you wanted to do `remove_wire(a, b)` it may only do it once, and the other one may remain.

Reply:
actually it's not liking a comparison between PortReferences